language/tools/move-package/src/lib.rs

```diff
@@ -168,21 +166,17 @@ impl BuildConfig {
         writer: &mut W,
     ) -> Result<CompiledPackage> {
         let resolved_graph = self.resolution_graph_for_package(path, writer)?;
-        let mutx = PackageLock::lock();
-        let ret = BuildPlan::create(resolved_graph)?.compile_no_exit(writer);
-        mutx.unlock();
-        ret
+        let _mutx = PackageLock::lock();
+        BuildPlan::create(resolved_graph)?.compile_no_exit(writer)
     }
 
     #[cfg(feature = "evm-backend")]
     pub fn compile_package_evm<W: Write>(self, path: &Path, writer: &mut W) -> Result<()> {
         // resolution graph diagnostics are only needed for CLI commands so ignore them by passing a
         // vector as the writer
         let resolved_graph = self.resolution_graph_for_package(path, &mut Vec::new())?;
-        let mutx = PackageLock::lock();
-        let ret = BuildPlan::create(resolved_graph)?.compile_evm(writer);
-        mutx.unlock();
-        ret
+        let _mutx = PackageLock::lock();
+        BuildPlan::create(resolved_graph)?.compile_evm(writer)
```

Comment:
What are the semantics for dropping unused locals like this in Rust? Are they well defined enough to rely on this?

Reply:
Yes, it's well defined -- drops get called in reverse order of declaration.  The gotchas are related to moves and the _ pattern -- if you move into a variable, you drop what was in the variable before and if you use the _ pattern instead of a variable name (tempting seeing as you don't need to refer to the guard), the guard will be dropped at assignment because it's treated as being never assigned.

Reply:
Here's a good resource that runs through the drop rules: https://vojtechkral.github.io/blag/rust-drop-order/